client/src/pages/Dashboard.js

```diff
@@ -16,27 +17,60 @@ const useStyles = makeStyles(() => ({
 
 const Dashboard = () => {
   const classes = useStyles();
+  const history = useHistory();
+
+  const [user, setUser] = useState({});
+  const [signedIn, setSignedIn] = useState(false);
+
+  useEffect(async () => {
+    checkCurrentUser();
+  }, []);
+
+  const checkCurrentUser = async () => {
+    //redirect to landing page if no token
+    let results = await fetch("/user/currentUser");
+
+    results = await results.json();
+
+    if (results.user === undefined) {
```

Comment:
Because `undefined` is a falsy value (it will be coerced into `false`), so you could do: 
```javascript
if (!results.user) { ... }
```